src/prover.rs

```diff
@@ -63,7 +63,7 @@ impl<E: Pairing, QAP: R1CSToQAP> Groth16<E, QAP> {
         let h_assignment = cfg_into_iter!(h)
             .map(|s| s.into_bigint())
             .collect::<Vec<_>>();
-        let h_acc = E::G1::msm_bigint(&pk.h_query, &h_assignment);
+        let h_acc = E::G1::msm_bigint(&pk.h_query, &h_assignment[..h_assignment.len() - 1]);
```

Comment:
This broke the tests in https://github.com/arkworks-rs/circom-compat/pull/68

@Pratyush I think the reasoning behind this change was that `pk.h_query` typically has `domain - 1` elements right? In which case `h_assignment` is of length = domain, but the last element is 0.

For some reason, the proving key imported from a `.zkey` file has length = domain. I'm going to investigate later, but I'm dropping the info here as I go along, maybe you have a clue as for this difference.

cc @jbaylina

Reply:
I think we would need to change to use `msm_unchecked`, or something that doesn't check length equality, because the arkworks Groth16 is of length domain - 1. This is because $z_a * z_b - z_c$ is of degree at most 2 * domain - 2, and so $h = (z_a * z_b - z_c)/v_H$ is of degree at most domain - 2 and hence requires domain - 1 coefficients.

Regarding what to replace `msm_bigint` with, I don't remember if `msm_bigint` panics when the input lengths are equal or not...

Reply:
`msm_bigint` works fine, it just takes the min length of the two. In arkworks-groth witness calculation this will be `domain-1` but in circom, both inputs will be of length `domain`.